packages/agents/src/agents/Evo/Evo.ts

```diff
@@ -1,160 +1,72 @@
 import {
-  LlmApi,
   Chat,
-  Workspace,
   Scripts,
-  Env,
   WrapClient,
   agentPlugin,
-  Logger,
   Timeout,
   basicFunctionCallLoop,
   AgentOutput,
   RunResult,
   AgentVariables,
 } from "@evo-ninja/agent-utils";
-import { AgentBase, AgentBaseConfig } from "../../AgentBase";
+import { AgentBase, AgentBaseContext } from "../../AgentBase";
 import {
-  DataAnalystAgent,
-  DeveloperAgent,
-  ResearcherAgent,
   ScriptedAgentOrFactory,
   ScriptedAgentContext,
 } from "../../scriptedAgents";
 import { DelegateAgentFunction } from "../../functions/DelegateScriptedAgent";
 import { OnGoalAchievedFunction } from "../../functions/OnGoalAchieved";
 import { OnGoalFailedFunction } from "../../functions/OnGoalFailed";
-import { Scripter } from "../Scripter";
 import { ResultErr } from "@polywrap/result";
 import { VerifyGoalAchievedFunction } from "../../functions/VerifyGoalAchieved";
+import { getDefaultDelegatedAgents } from "./getDefaultDelegatedAgents";
+import * as prompts from "./prompts";
+
+const AGENT_NAME = "Evo";
 
 export interface EvoRunArgs {
   goal: string
 }
 
 export class Evo extends AgentBase<EvoRunArgs, ScriptedAgentContext> {
   constructor(
-    llm: LlmApi,
-    chat: Chat,
-    logger: Logger,
-    workspace: Workspace,
+    context: AgentBaseContext,
     scripts: Scripts,
-    env: Env,
     timeout?: Timeout,
-    scriptedAgents?: ScriptedAgentOrFactory[]
+    delegatedAgents?: ScriptedAgentOrFactory[]
```

Comment:
Couldn't these potentially be just agents and not scripted agents?

Reply:
The problem here exists within the "Agent Args". The scripted agents all have "goal" & "context" as their args. I think we should probably standardize this for all agents, which would allow us to unify them.

Reply:
What he said :D